core/gather/gatherers/trace-elements.js

```diff
@@ -65,6 +65,61 @@ class TraceElements extends BaseGatherer {
     if (name) this.animationIdToName.set(id, name);
   }
 
+  /**
+   * @param {LH.Artifacts.TraceEngineResult} traceEngineResult
+   * @param {string} mainFrameId
+   * @return {Promise<Array<{nodeId: number}>>}
+   */
+  static async getTraceEngineElements(traceEngineResult, mainFrameId) {
+    // Can only resolve elements for the latest insight set, which should correspond
+    // to the current frame id. Can't resolve elements for pages that are gone.
+    const insightSet = [...traceEngineResult.insights.values()]
+      .reverse().find(insightSet => insightSet.frameId === mainFrameId);
```

Comment:
Pretty sure this will always be the last insight set in the list. The main frame ID does not change across navigations.